### PR TITLE
Improved Compiler Error Messages

### PR DESCRIPTION
**Pull request by Genaro-Chris**
Include the proposal  to improve Swift error messages

### CODE REVIEW

File: proposals/nnnn-improved-compiler-error-messages.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+HelloWorld.cs(8,8): error CS1061: Type `Person' does not contain a definition for `walk' and no extension method `walk' of type `Person' could be found. 
+Are you missing an assembly reference?
+HelloWorld.cs(13,14): (Location of the symbol related to previous error)
+```
+The error codes is hosted online at [C# Compiler messages](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/).
+
+
+While that of Rust language
+
+```
```
Comment (PsychoH13) [inline proposals/nnnn-improved-compiler-error-messages.md:75]:
```suggestion
```rust
```

File: proposals/nnnn-improved-compiler-error-messages.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+This proposal would help categorize error messages, explain the cause of the error and possibly with code examples, ways of preventing or correcting such errors from recurring.
+ 
+## Proposed solution
+
+Taking similar routes as some other languages such as Rust or C# as such languages as grouped error messages into some sort of error message index for easy referencing, tips for prevention and quick look-ups.
+
+Example with that of C# language
+
+```
```
Comment (PsychoH13) [inline proposals/nnnn-improved-compiler-error-messages.md:39]:
```suggestion
```csharp
```

File: proposals/nnnn-improved-compiler-error-messages.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+The error code index is hosted online at [Rust error codes index](https://doc.rust-lang.org/error_codes/error-index.html)
+
+## Detailed design
+
+Still using our Swift example, the error should look like this instead
+
+```
+Found error SE0157: value of type 'Person' has no member 'walk
+  adam.walk()
+	~~~~ ^ ~~~~
```
Comment (PsychoH13) [inline proposals/nnnn-improved-compiler-error-messages.md:113]:
```suggestion
    ~~~~ ^~~~
```

File: proposals/nnnn-improved-compiler-error-messages.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+* Authors: [Genaro-Chris](https://github.com/Genaro-Chris)
+* Review Manager: TBD
+* Status: **Awaiting implementation** or **Awaiting review**
+* Review: ([Pitch](https://forums.swift.org/t/pitch-improved-compiler-error-messages/66839))
+
+## Introduction
+
+Currently the Swift compiler error messages aren't really helping convene the true problem encountered during compilation. Sometimes either the message displayed by the compiler are sometimes generic or not helpful in trying to aid the programmer correct him or herself, or errors hard to recreate in a different context or situation. This often leads to quick dismay of our sweet dear language by some new developers coming from other languages.
+
+```
```
Comment (PsychoH13) [inline proposals/nnnn-improved-compiler-error-messages.md:13]:
```suggestion
```swift
```

File: proposals/nnnn-improved-compiler-error-messages.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+Found error SE0157: value of type 'Person' has no member 'walk
+  adam.walk()
+	~~~~ ^ ~~~~
+		       
+For more information about this error, try swiftc --describe SE0157	
+```
+Also using a new compiler flag 'describe', we could reference the preceding error category
+
+```shell
+$swift describe SE0157
```
Comment (PsychoH13) [inline proposals/nnnn-improved-compiler-error-messages.md:120]:
```suggestion
$ swift describe SE0157

```

**Comment by Jumhyn** [comment]
Hi @Genaro-Chris! As mentioned in the pitch thread you've linked, improvements to compiler diagnostics are not generally subject to Swift Evolution and would not require a proposal. It also looks like there hasn't been discussion on the forums regarding this topic for several months, unless there's a thread I'm missing? I would recommend trying to push the discussion forward on the forums if you'd like to see motion on this topic.

File: proposals/nnnn-improved-compiler-error-messages.md (hunk 5 above)
Comment (Genaro-Chris) [reply proposals/nnnn-improved-compiler-error-messages.md:120]: Hey @PsychoH13, thanks for all the corrections

**Comment by Genaro-Chris** [comment]
Hey @Jumhyn, so what could you suggest I do keep pushing it on the forums first?

**Comment by rjmccall** [comment]
Freddy is right: this isn't a topic for evolution, and I need to close this PR.

One way you could get more engagement would be to start developing the idea in specific detail.  For example, you could go through some of the current diagnostics in the compiler and talk about how you would categorize them.

Mostly, though, this is a request for work.  If you want to see it happen, you may need to do some of the work yourself.  Diagnostic emission is a relatively self-contained subsystem in the compiler; you don't have to be able to rewrite the generics system to add a category field to the diagnostics tables.

**Pull request closed without merging**